jupyter_client/session.py

```diff
@@ -110,14 +110,33 @@ def squash_unicode(obj):
 # Mixin tools for apps that use Sessions
 #-----------------------------------------------------------------------------
 
+def uuid4():
+    """Generate a new UUID.
+
+    Avoids problematic runtime import in stdlib uuid on Python 2.
+
+    Returns
+    -------
+
+    uuid string (16 random bytes as hex-encoded unicode str)
+    """
+    buf = os.urandom(16)
+    return u'-'.join(b2a_hex(x).decode('ascii') for x in (
+        buf[:4], buf[4:6], buf[6:8], buf[8:10], buf[10:]
```

Comment:
Even a random UUID is not purely random bytes; there are a couple of type fields in there that mark what kind of UUID it is:

https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_.28random.29

As you say, we don't really need UUIDs, just a string that's unique in context. So if we're just generating some random bytes, that's OK, but let's not call it a UUID or format it like a UUID.